static/src/javascripts/projects/commercial/modules/dfp/dfp-api.spec.js

```diff
@@ -0,0 +1,422 @@
+            section: 'news',
+            seriesId: 'learning/series/happy-times',
+            sharedAdTargeting: {
+                ct: 'Article',
+                edition: 'us',
+                k: ['korea', 'ukraine'],
+                se: ['happy-times'],
+            },
+        };
+        config.images = {
+            commercial: {},
+        };
+        config.ophan = {
+            pageViewId: 'dummyOphanPageViewId',
+        };
+
+        if (document.body) {
+            document.body.innerHTML = domSnippet;
+        }
+
+        $style = $.create('<style type="text/css"></style>')
+            .html(`body:after{ content: "wide"}`)
+            .appendTo('head');
+        const pubAds = {
+            listeners: [],
+            addEventListener: jest.fn(function(eventName, callback) {
+                this.listeners[eventName] = callback;
+            }),
+            setTargeting: jest.fn(),
+            enableSingleRequest: jest.fn(),
+            collapseEmptyDivs: jest.fn(),
+            refresh: jest.fn(),
+        };
+        const sizeMapping = {
+            sizes: [],
+            addSize: jest.fn(function(width, sizes) {
+                this.sizes.unshift([width, sizes]);
+            }),
+            build: jest.fn(function() {
+                const tmp = this.sizes;
+                this.sizes = [];
+                return tmp;
+            }),
+        };
+        window.googletag = {
+            cmd: {
+                push(...argz) {
+                    const args = Array.prototype.slice.call(argz);
+                    args.forEach(command => {
+                        command();
+                    });
+                },
+            },
+            pubads() {
+                return pubAds;
+            },
+            sizeMapping() {
+                return sizeMapping;
+            },
+            defineSlot: jest.fn(() => window.googletag),
+            defineOutOfPageSlot: jest.fn(() => window.googletag),
+            addService: jest.fn(() => window.googletag),
+            defineSizeMapping: jest.fn(() => window.googletag),
+            setTargeting: jest.fn(() => window.googletag),
+            enableServices: jest.fn(),
+            display: jest.fn(),
+        };
+        // eslint-disable-next-line no-underscore-dangle
+        window.__switch_zero = false;
+
+        commercialFeatures.dfpAdvertising = true;
+
+        done();
+    });
+
+    afterEach(() => {
+        reset();
+        if (document.body) {
+            document.body.innerHTML = '';
+        }
+        $style.remove();
+        window.googletag = null;
+    });
+
+    it('should exist', () => {
+        expect(prepareGoogletag).toBeDefined();
+        expect(getAdverts).toBeDefined();
+        expect(getCreativeIDs).toBeDefined();
+    });
+
+    it('hides all ad slots when all DFP advertising is disabled', done => {
+        commercialFeatures.dfpAdvertising = false;
+        new Promise(resolve => {
```

Comment:
I think you could just return the Promise instead of the

```js
    .then(done)
    .catch(done.fail);
```

magic. What do you think?